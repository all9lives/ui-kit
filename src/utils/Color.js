export const invert = (hex, bw) => {
  if (hex.indexOf('#') === 0) {
    hex = hex.slice(1)
  }
  if (hex.length === 3) {
    hex = hex[0] + hex[0] + hex[1] + hex[1] + hex[2] + hex[2]
  }
  if (hex.length !== 6) {
    throw new Error('Invalid HEX color.')
  }
  let r = parseInt(hex.slice(0, 2), 16)
  let g = parseInt(hex.slice(2, 4), 16)
  let b = parseInt(hex.slice(4, 6), 16)
  if (bw) {
    return (r * 0.299 + g * 0.587 + b * 0.114) > 186 ? '#000000' : '#FFFFFF'
  }
  r = (255 - r).toString(16)
  g = (255 - g).toString(16)
  b = (255 - b).toString(16)
  return `#${r.padStart(2, '0')}${g.padStart(2, '0')}${b.padStart(2, '0')}`
}

export default {
  primary: '#b71c1c',
  primary_light: '#ea4949',
  error: '#f44336',
  warning: '#f2b233',
  success: '#2ac686',
  black: '#191919',
  white: '#fafafa',
  gray: '#cacaca'
}
