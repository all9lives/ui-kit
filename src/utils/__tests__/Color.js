/* eslint-env jest */
import { invert } from '../Color'

describe('Color', () => {
  it('inverts hex color correctly', () => {
    const result = invert('#282828')
    expect(result).toMatchSnapshot()
  })

  it('inverts 3 digits hex color correctly', () => {
    const result = invert('#ccc')
    expect(result).toMatchSnapshot()
  })

  it('inverts hex color even without hash', () => {
    const result = invert('ccc')
    expect(result).toMatchSnapshot()
  })

  it('inverts black and white correctly', () => {
    const result = invert('fff', true)
    expect(result).toMatchSnapshot()
  })

  it('does not handle incorrect digits hex color', () => {
    try {
      invert('#1234567')
    } catch (e) {
      expect(e).toMatchSnapshot()
    }
  })
})
