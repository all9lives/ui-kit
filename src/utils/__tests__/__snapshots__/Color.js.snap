// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color does not handle incorrect digits hex color 1`] = `[Error: Invalid HEX color.]`;

exports[`Color inverts 3 digits hex color correctly 1`] = `"#333333"`;

exports[`Color inverts black and white correctly 1`] = `"#000000"`;

exports[`Color inverts hex color correctly 1`] = `"#d7d7d7"`;

exports[`Color inverts hex color even without hash 1`] = `"#333333"`;
